content/facets-endpoint/index.md

```diff
@@ -0,0 +1,93 @@
+---
+rfc: 0022
+start_date: 2018-09-28
+decision_date:
+pr: openregisters/registers-rfc#39
+status: draft
+---
+
+# Facets endpoint
+
+## Summary
+
+This RFC proposes to change the endpoint for the records' “facets” to simplify
+path handling.
+
+## Motivation
+
+Currently the REST API for facets uses a path structure that overlaps with the
+standard record structure. This makes path handling (e.g. analytics, logging)
+harder than it needs to be. Observation suggests that users can get confused
+of what is the resource about.
+
+
+## Explanation
+
+For context, the current endpoint for facets is:
+
+```
+GET /records/{attribute-name}/{attribute-value}
+```
+
+The record endpoint is:
+
+```
+GET /records/{key}
+```
+
+And the records endpoint is:
+
+```
+GET /records
+```
+
+
+This RFC proposes to change the facets endpoint to extend the records endpoint
+with a query string to make more obvious that a “facet” is filter on the
+records collection.
+
+```
+GET /records{?name, value}
+```
+
+Where `{?name}` is required if `{?value}` is provided.
+
+
+### Filter by strict value match
+
+```
+GET /records?name={attribute-name}&value={attribute-value}
+```
+
+Exactly the current behaviour.
+
+### Filter by name
+
+```
+GET /records?name={attribute-name}
```

Comment:
to clarify, I think this is new functionality there is no endpoint for this in the current API. I guess there is a tension with the filtering capabilities we should provide v.s. what clients should implement (as we saw with the indexes functionality). So I am not sure where we should be going with this but this particular one does feel logical as it feels like a subset of the facetted value query, just allowing for non-null. 
If there is more debate about this one option would be to move this new query into a separate RFC and make this one just about the new route/redirects.

Reply:
Just to clarify, I agree with the motivation for this RFC, my concern is just that the choice of URL structure will bake in certain constraints, and we won't be able to add certain kinds of functionality without making another breaking change later on.

Whether this is a big deal or not depends on how likely we are to actually do these kinds of changes, and how happy we would be to make breaking changes in this area of the API. I'm lacking the bigger picture here so I'm happy to defer to your judgement on this.

I don't think we need to split the RFC unless we want to have a bigger discussion about the facets functionality now?

Reply:
I have removed this part of the RFC to handle the null vs non-null conversation separated from the core intention of this proposal.